### DISCUSSION TITLE
using File Manger and printers when client connected via PiVPN?

### DISCUSSION THREAD

**Discussion by jfabernathy** [pivpn/pivpn#1673] [Q&A]
I have a working PiHole and PiVPN system on a RPi 3B+.  Works great.  I'm can connected to all my PCs and servers in my home using the PiVPN and the only thing not working is Printing to my Network printers on the home network and using the File Manager to access my cifs/smb NAS on my home network.  If I use the terminal on my client and mount the NAS using the mount command I can use the terminal to copy files from the NAS.

Is there a way to make FileManger and the printers work when on VPN?

**Comment by Pleonazm** [comment]
You did not give many details, but I will try.
I assume you have a network printer with a static IP, and use Wireguard.
Add PRI.NT.ER.IP/32 route to Allowed IPs in all clients (which need access to the printer) files, and in the server's file in the section for the host with the printer connected to.
Then add your printer directly by IP, not things like network shares.

Note this will obviously cause conflicts if the same IP address is on the client's subnet.

**Reply by jfabernathy** [reply]
Actually I use OpenVPN and not Wireguard. Are the instructions similar?

**Reply by Pleonazm** [reply]
The logic should be the same.
IIRC the OpenVPN allows pushing additional routes directly to the clients.

https://forums.openvpn.net/viewtopic.php?t=9055
